### ISSUE TITLE
Change 2000 to 100 in aws_waf_rate_based_rule validation

### ISSUE THREAD

**Issue by vladgolubev** [terraform-providers/terraform-provider-aws#10058]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.6

### Affected Resource(s)

* aws_wafregional_rate_based_rule
* aws_waf_rate_based_rule

### Terraform Configuration Files

```hcl
resource "aws_wafregional_rate_based_rule" "rate_limit" {
  name        = "${var.waf_prefix}-generic-rate-limit"
  metric_name = "${var.waf_prefix}ratelimit"
  rate_key    = "IP"
  rate_limit  = 100 # <== Note value 100 here

  predicate {
    data_id = aws_wafregional_byte_match_set.some_endpoint.id
    negated = false
    type    = "ByteMatch"
  }
}
```

### Expected Behavior

See: https://aws.amazon.com/about-aws/whats-new/2019/08/lower-threshold-for-aws-waf-rate-based-rules/

> The new threshold of 100 requests per 5 minutes (previously 2000 requests per 5 minutes) gives you greater control ....

### Actual Behavior

		

```
Error: expected rate_limit to be at least (2000), got 100
```

### Steps to Reproduce

1. `terraform apply`

**Comment by jeet-parekh** [comment]
Looks like a good first issue. Can I work on this?

**Comment by vladgolubev** [comment]
Looks like closed by #9946?

**Comment by bflad** [comment]
Hi folks 👋 This is not directly closed with #9946, which only addressed the validation within the Terraform schema and did not update our acceptance testing to double check the lower values actually worked. AWS Go SDK v1.24.4 (current as of this writing) is still requiring the rate limit to be greater than or equal to 2000 for both WAF and WAF Regional services.

The fix will need to be completed by the AWS WAF service team to change the API model from which the Go SDK is generated or by the AWS Go SDK team to fix the SDK generation. I have created an upstream issue for tracking: https://github.com/aws/aws-sdk-go/issues/2851

I would suggest following there and submitting AWS Support case or talking with your AWS TAM if you would like to expedite the remediation process.

**Comment by bflad** [comment]
It looks like this is fixed in AWS Go SDK v1.25.2: https://github.com/aws/aws-sdk-go/pull/2862/files#diff-bbf2caec0ea70a2df0e6875ce78c0003R10303

This issue will be updated when that update is merged and scheduled for release in the provider.

**Comment by bflad** [comment]
Hi again 👋 

This was resolved in version 2.31.0 of the Terraform AWS Provider, released earlier today. 👍 

```hcl
provider "aws" {
  region  = "us-east-2"
  version = "2.31.0"
}

resource "aws_wafregional_rate_based_rule" "test" {
  name        = "test"
  metric_name = "test"
  rate_key    = "IP"
  rate_limit  = 100
}
```

```console
$ terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_wafregional_rate_based_rule.test will be created
  + resource "aws_wafregional_rate_based_rule" "test" {
      + id          = (known after apply)
      + metric_name = "test"
      + name        = "test"
      + rate_key    = "IP"
      + rate_limit  = 100
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_wafregional_rate_based_rule.test: Creating...
aws_wafregional_rate_based_rule.test: Creation complete after 2s [id=bcec126b-7fde-450f-b89f-e1d6d9455d02]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

For further bug reports or feature requests with this functionality, please create a new GitHub issue following one of the templates. Thanks!
